cuda/cuda_config.h missing when compiling custom ops with nvcc

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
Yes, see below.
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Ubuntu 16.04
- **TensorFlow installed from (source or binary)**:
Source
- **TensorFlow version (use command below)**:
v1.3.0-0-g9e76bf324 1.3.0
- **Python version**: 
3.5.2
- **Bazel version (if compiling from source)**:
0.5.4
- **CUDA/cuDNN version**:
8.0.44 / 5.1.5
- **GPU model and memory**:
Any.
- **Exact command to reproduce**:
See below.

### Describe the problem

When compiling a custom op using nvcc, which includes `tensorflow/core/util/cuda_kernel_helper.h`, I get the following error:

```
/usr/local/cuda-8.0/bin/nvcc -c -o ~/Code/libspn/build/ops/gather_columns_functor_gpu.cu.cc.o ~/Code/libspn/libspn/ops/gather_columns_functor_gpu.cu.cc -std=c++11 -x=cu -Xcompiler -fPIC -DGOOGLE_CUDA=1 --expt-relaxed-constexpr -I ~/.local/lib/python3.5/site-packages/tensorflow/include -gencode=arch=compute_35,"code=sm_35,compute_35" -gencode=arch=compute_52,"code=sm_52,compute_52" -gencode=arch=compute_61,"code=sm_61,compute_61"
In file included from ~/.local/lib/python3.5/site-packages/tensorflow/include/tensorflow/core/platform/default/stream_executor.h:26:0,
                 from ~/.local/lib/python3.5/site-packages/tensorflow/include/tensorflow/core/platform/stream_executor.h:24,
                 from ~/.local/lib/python3.5/site-packages/tensorflow/include/tensorflow/core/util/cuda_kernel_helper.h:26,
                 from ~/Code/libspn/libspn/ops/gather_columns_functor_gpu.cu.h:11,
                 from ~/Code/libspn/libspn/ops/gather_columns_functor_gpu.cu.cc:5:
~/.local/lib/python3.5/site-packages/tensorflow/include/tensorflow/stream_executor/dso_loader.h:32:30: fatal error: cuda/cuda_config.h: No such file or directory
compilation terminated.
```

Copying `cuda_config.h` to `/site-packages/tensorflow/include/tensorflow/stream_executor/cuda` solves the problem.

The same issue has been observed by several other users in #6602 (see the comments added after the issue was closed).

The commit 2c598e874e6a7b6b3185846ce9bac97a7d5d0169 destroys my makefile, too. This commit changed consists of several

```diff
-#include "third_party/gpus/cuda/include/cuda.h"
+#include "cuda/include/cuda.h"
```

giving the exact same error message:

```
../site-packages/tensorflow/include/tensorflow/core/util/cuda_kernel_helper.h:28:80: fatal error: cuda/include/cuda.h: No such file or directory
compilation terminated.
```

compiling user-ops with GPU does not work since 1.3.1.

Does anybody know where tensorflower-gardener currently lifes? He is the guy who usually introduces the breaking changes!

@PatWie +1
I encounter this problem when build android tensorflow_demo, giving the following error message:
```bash
tensorflow/core/kernels/lrn_op.cc:34:10: fatal error: 'cuda/include/cuda.h' file not found
```

With TF 1.3, I only find `cuda_config.h` missing, but when compiling against current master 1.4-rc0/1, I also additionally get:
```
tensorflow/include/tensorflow/core/util/cuda_kernel_helper.h:24:31: fatal error: cuda/include/cuda.h: No such file or directory
compilation terminated.
```

I also encounter the same problem.
`/usr/local/lib/python3.5/dist-packages/tensorflow/include/tensorflow/stream_executor/dso_loader.h:32:30: fatal error: cuda/cuda_config.h: No such file or directory`

--- a/tensorflow/core/common_runtime/gpu/gpu_device.cc
+++ b/tensorflow/core/common_runtime/gpu/gpu_device.cc
@@ -65,6 +65,10 @@ limitations under the License.
 #include "tensorflow/core/util/env_var.h"
 #include "tensorflow/core/util/stream_executor_util.h"
 
+#if !defined(PLATFORM_GOOGLE)
+#include "cuda/cuda_config.h"
+#endif
+
 namespace tensorflow {
 
 // Eigen Ops directly allocate memory only for temporary buffers used

--- a/tensorflow/stream_executor/dso_loader.cc
+++ b/tensorflow/stream_executor/dso_loader.cc
@@ -33,6 +33,10 @@ limitations under the License.
 #include "tensorflow/stream_executor/platform/logging.h"
 #include "tensorflow/stream_executor/platform/port.h"
 
+#if !defined(PLATFORM_GOOGLE)
+#include "cuda/cuda_config.h"
+#endif
+
 namespace perftools {
 namespace gputools {
 namespace internal {

--- a/tensorflow/stream_executor/dso_loader.h
+++ b/tensorflow/stream_executor/dso_loader.h
@@ -28,10 +28,6 @@ limitations under the License.
 #include "tensorflow/stream_executor/platform.h"
 #include "tensorflow/stream_executor/platform/mutex.h"
 
-#if !defined(PLATFORM_GOOGLE)
-#include "cuda/cuda_config.h"
-#endif
-
 namespace perftools {
 namespace gputools {
 namespace internal {

